### ISSUE TITLE
Unexpected valid-jsdoc duplicate JSDoc parameter

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#3476]
```js
/**
 * Finds all different day times across a range of timestamps.
 * Sorts day times in an ascending order.
 *
 * @param {Object[]} screenings Array of screenings.
 * @param {Number} screenings[].timestamp
 * @return {Array}
 */
```

This is a common syntax to define member properties of the object. JSDoc validator is raising an error: "Duplicate JSDoc parameter 'screenings'".

```
curiosity:foo gajus$ eslint -v
v1.2.1
```

```
/foo/index.js
  14:5   error  Duplicate JSDoc parameter 'screenings'                                  valid-jsdoc
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by gajus** [comment]
http://stackoverflow.com/a/32139970/368691

**Comment by BYK** [comment]
According to http://usejsdoc.org/tags-param.html#parameters-with-properties your code looked valid and I was about to accept it :D

**Comment by gajus** [comment]
I was not aware of the `@typedef` at the time of raising this issue. I consider the following syntax a lot more eloquent:

```js
/**
 * @typedef {Object} screening
 * @property {Number} timestamp UNIX timestamp.
 * @property {String} url Booking URL.
 */

/**
 * @param {screening[]}
 */
function positionTimes (screenings) {}
```

**Comment by gajus** [comment]
Though you are right. It is a valid syntax and should be supported.

**Comment by gajus** [comment]
There is an open issue regarding this in the doctrine, https://github.com/Constellation/doctrine/issues/111.

**Comment by gajus** [comment]
This was bothering me a lot. I have written a dirty fix that patches "valid-jsdoc.js". https://gist.github.com/gajus/e5858033042a8ac6d476/revisions#diff-419160ac55743894fd8b0a337e62db22R110 Sharing in case some one else needs it.

Doctrine repository has not pulled any PRs in several months. Does not look welcoming.

Might be worth considering using https://www.npmjs.com/package/jsdoc-parse instead.

**Comment by nzakas** [comment]
Thanks, we can definitely look at switching parsers. We'd just need someone to carve out the time to do it.

**Comment by nzakas** [comment]
This should be unblocked due to Doctrine 0.7.0 released. Anyone want to take this on?
